### ISSUE TITLE
Load whole models

### ISSUE THREAD

**Issue by Rainer-Lang** [Raizlabs/DBFlow#154]
I have some models which includes also others:
public List<E>...

So I want to load all the models with all included lists.

How can this be done?

**Comment by agrosner** [comment]
if you have a `Model` which lazy-loads `List` data, have your model implement `LoadFromCursorListener` and in the `onLoadFromCursor()` method call the lazy-loaders. that will give you all of the data on-load. Also I am considering adding an annotation to enable these kind of methods to load automatically.

**Comment by Rainer-Lang** [comment]
Thank you very much! :-) 
Is this also described in the wiki or included in the project?

**Comment by Rainer-Lang** [comment]
I'm very interested in the annotation. Do you or anybody else have an example?

**Comment by agrosner** [comment]
apparently not yet. Let me add those when I find time. Ill mark this to resolve.

**Comment by tsuharesu** [comment]
What I do here is lazy-load the List when I need it. It would be something like
```
class Person extends BaseModel {
    private List<Toy> toys;

    public List<Toy> getToys() {
        if (toys == null)
            toys = new Select().from(Toy.class)// the rest of the select here
        return toys
    }
}
```

So, when I need the toys, I just use `person.getToys()`. If the list is already fetched, I get it from "memory". If not, Select will fetch it and the returns.

**Comment by Rainer-Lang** [comment]
@tsuharesu But if you are loading/select a Person the toys aren't selected/loaded also.
I want to select Person and also ALL other models inside.

**Comment by agrosner** [comment]
```java

class Person extends BaseModel implements LoadFromCursorListener {
    private List<Toy> toys;

    public List<Toy> getToys() {
        if (toys == null)
            toys = new Select().from(Toy.class)// the rest of the select here
        return toys
    }

@Override
public void onLoadFromCursor(Cursor cursor) {
  getToys();
}
}
```

What I'm saying is I may add an annotation like:

```java

@Call(methods = {LOAD, DELETE})
public List<Toy> getToys() {
        if (toys == null)
            toys = new Select().from(Toy.class)// the rest of the select here
        return toys
    }

```

Which could call this method in the generated class on load, deletion, save, etc... so it's more managed and less manual.

**Comment by Rainer-Lang** [comment]
@agrosner The annotation would be really helpful - for all (load/delete/save/..)!
Can't wait to get this. ;-)

**Comment by Rainer-Lang** [comment]
@agrosner Any news? Integrated?

**Comment by Rainer-Lang** [comment]
@agrosner I really need this.
This issue was closed but added to milestone 2.0.0
